fabtests/common/hmem_cuda.c

```diff
@@ -192,12 +240,45 @@ int ft_cuda_init(void)
 		goto err_dlclose_cuda;
 	}
 
+#if HAVE_CUDA_DMABUF
+	cuda_ops.cuMemGetHandleForAddressRange = dlsym(cuda_handle,
+						       STRINGIFY(cuMemGetHandleForAddressRange));
+	if (!cuda_ops.cuPointerSetAttribute) {
+		FT_ERR("Failed to find cuMemGetHandleForAddressRange\n");
+		goto err_dlclose_cuda;
+	}
+#endif
+	cuda_ops.cuDeviceGetAttribute = dlsym(cuda_handle,
+					      STRINGIFY(cuDeviceGetAttribute));
+	if (!cuda_ops.cuPointerSetAttribute) {
+		FT_ERR("Failed to find cuPointerSetAttribute\n");
+		goto err_dlclose_cuda;
+	}
+
+	cuda_ops.cuDeviceGet = dlsym(cuda_handle,
+				     STRINGIFY(cuDeviceGet));
+	if (!cuda_ops.cuPointerSetAttribute) {
+		FT_ERR("Failed to find cuDeviceGet\n");
+		goto err_dlclose_cuda;
+	}
+
+	cuda_ops.cuMemGetAddressRange = dlsym(cuda_handle,
+				     STRINGIFY(cuMemGetAddressRange));
+	if (!cuda_ops.cuMemGetAddressRange) {
+		FT_ERR("Failed to find cuMemGetAddressRange\n");
+		goto err_dlclose_cuda;
+	}
+
 	cuda_ret = cuda_ops.cudaSetDevice(opts.device);
 	if (cuda_ret != cudaSuccess) {
 		CUDA_ERR(cuda_ret, "cudaSetDevice failed");
 		goto err_dlclose_cuda;
 	}
 
+	ret = ft_cuda_hmem_detect_dmabuf_support();
+	if (ret != FI_SUCCESS)
+		goto err_dlclose_cuda;
```

Comment:
Do you want to always fail the test when dmabuf support is not available even if the test doesn't need dmabuf?

Reply:
`ft_cuda_hmem_detect_dmabuf_support()` still returns SUCCESS when dmabuf support is not available. It will return non-zero when there is some cuda API failure

Reply:
Nah I have a bug here. Updated, thanks for catching it